src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -129,5 +129,32 @@ internal static bool IsPrimitiveType(this CorElementType et)
         [Intrinsic]
         internal static bool IsKnownConstant(int t) => false;
 #pragma warning restore IDE0060
+
+#if NATIVEAOT
+        [System.Runtime.RuntimeExport("RhRuntimeHelpers_MemSet")]
+#endif
+        private static unsafe void MemSet(ref byte dest, byte value, nuint size)
+        {
+            if (size > 0)
+            {
+                // Implicit nullcheck
+                _ = Unsafe.ReadUnaligned<byte>(ref dest);
+                SpanHelpers.Fill(ref dest, size, value);
```

Comment:
Is this going to compile into tailcall once both your changes are in, or are we going to take the regression from https://github.com/MihuBot/runtime-utils/issues/258#issue-2145007226 ?

Reply:
Unfortunately not 😞 methods marked with [Intrinsic] aren't tail called. If I fix it, I will then won't be able to unroll unrollable intrinsics if they're tail-called.

We have two options:
1) Accept this regression. Actually, I think it should be still faster now because I introduced a faster `CORINFO_HELP_MEMZERO` so this path is only ever invoked for explicit `initblk` opcode with non-zero value. (in C# it's basically only via Unsafe.InitBlock api). 

2) Inline Fill here (code duplication).

Same for MemCpy.